src/pymor/reductors/data_driven.py

```diff
@@ -0,0 +1,268 @@
+                                                  parameters=self.parameters, output_functional=self.output_functional,
+                                                  input_scaler=self.input_scaler, output_scaler=self.output_scaler,
+                                                  time_vectorized=self.time_vectorized)
+        return rom
+
+    def extend_training_data(self, parameters, snapshots):
+        """Add sequences of parameters and corresponding snapshots to the training data."""
+        self.training_data.extend(self._compute_data(parameters, snapshots))
+
+
+class DataDrivenPODReductor(DataDrivenReductor):
+    """Reductor building a reduced basis and relying on a machine learning surrogate.
+
+    In addition to the :class:`~pymor.reductors.data_driven.DataDrivenReductor`,
+    this reductor uses snapshot data in order to construct a reduced basis via POD
+    and projects the snapshots onto the reduced basis to generate data for the
+    machine learning training. The approach is described in :cite:`HU18`.
+    See :class:`~pymor.reductors.data_driven.DataDrivenReductor` for more details.
+
+    Parameters
+    ----------
+    training_parameters
+        |Parameter values| to use for training of the regressor.
+    training_snapshots
+        |VectorArray| to use for the training of the regressor.
+        Contains the solutions or outputs associated to the parameters in
+        `training_parameters`.
+        In the case of a time-dependent problem, the snapshots are assumed to be
+        equidistant in time.
+    regressor
+        See :class:`~pymor.reductors.data_driven.DataDrivenReductor`.
+    T
+        See :class:`~pymor.reductors.data_driven.DataDrivenReductor`.
+    time_vectorized
+        See :class:`~pymor.reductors.data_driven.DataDrivenReductor`.
+    output_functional
+        |Operator| mapping a given solution to the model output. In many applications,
+        this will be a |Functional|, i.e. an |Operator| mapping to scalars.
+        This is not required, however.
+        The output functional will be projected automatically onto the reduced space.
+    input_scaler
+        See :class:`~pymor.reductors.data_driven.DataDrivenReductor`.
+    output_scaler
+        See :class:`~pymor.reductors.data_driven.DataDrivenReductor`.
+    product
+        Inner product |Operators| defined on the discrete space the
+        problem is posed on. Used for reduced basis computation via POD and
+        or orthogonal projection onto the reduced basis.
+    pod_params
+        Dict of additional parameters for the POD-method.
+    """
+
+    def __init__(self, training_parameters, training_snapshots,
+                 regressor=VKOGARegressor(GaussianKernel()),
+                 T=None, time_vectorized=False, output_functional=None,
+                 input_scaler=None, output_scaler=None, product=None,
+                 pod_params={}):
```

Comment:
The same mutable default argument argumentation from above would apply here for the `pod_params` and the `regressor`.